### PR TITLE
Add Makefile for common tasks

### PR DESCRIPTION
**Pull request by michaelsauter**
Addresses #381.

Could be extended for other purposes, but this might be a good start.

Use `make help` to get an overview of available targets.

@gerardcl Maybe we can test-drive this next week? :)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,112 @@
+SHELL = /bin/bash
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
+MAKEFLAGS += --warn-undefined-variables
+MAKEFLAGS += --no-builtin-rules
```
Comment (henrjk) [inline Makefile]: The values look reasonable to me. Do you have a specific source of inspiration for them?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,112 @@
+SHELL = /bin/bash
```
Comment (henrjk) [inline Makefile]: Should this be `SHELL = /usr/bin/env bash` ?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	tailor export -n cd > backup_cd.yml
+.PHONY: backup-ocp-config
+
+## Create a backup of the SonarQube database in the current directory.
+backup-sonarqube:
+	cd sonarqube && sh backup.sh .
+.PHONY: backup-sonarqube
+
+# HELP
+# Based on https://gist.github.com/prwhite/8168133#gistcomment-2278355.
```
Comment (henrjk) [inline Makefile]: Cool to generate the help!

File: Makefile (hunk 1 above)
Comment (michaelsauter) [reply Makefile]: Takne from https://tech.davis-hansson.com/p/make/. It has a few other things that I don't think are helpful as e.g. they need a newer make version than available in MacOS Catalina by default.

File: Makefile (hunk 2 above)
Comment (michaelsauter) [reply Makefile]: Maybe - I never really care that much. Re-reading https://unix.stackexchange.com/questions/29608/why-is-it-better-to-use-usr-bin-env-name-instead-of-path-to-name-as-my, your suggestion might be better for portability. Did you ever encounter issues with scripts using just `/bin/bash`?

Comment (metmajer) [reply Makefile]: Personally, not with Bash. I have mainly seen this useful when using other interpreters, such as ruby or groovy.

**Pull request merged**